Refactor public methods to avoid in contract

daniel-zullo opened the pull request:
`in` contracts containing asserts should only be used on private
methods which are called via a public API including verify/enforce
checks as they will be completely removed from the program when
compiled in `-release` mode.

All checks in `in` contracts have been converted to verify statements
in the method body.

nemanja-boric-sociomantic commented on src/ocean/task/IScheduler.d:
```diff
@@ -307,12 +308,10 @@ public interface IScheduler
 {
+    verify(_scheduler !is null,
+        "Scheduler is null, initScheduler must be called before using it");
+
```
I think this one stay as is, though. Or change this `verify` to `assert`. This is the condition that can fail only once per runtime.

daniel-zullo replied:
you mean remove the `in` contract and keep the `assert()`, right?

nemanja-boric-sociomantic replied:
Yeah :+1:

Commit:
Refactor public methods to avoid in contract

`in` contracts containing asserts should only be used on private
methods which are called via a public API including verify/enforce
checks as they will be completely removed from the program when
compiled in `-release` mode.

All checks in `in` contracts have been converted to verify statements
in the method body excepts when the condition can fail only once per
runtime.

daniel-zullo commented:
Updated based on a suggestion from Nemanja.

<details>
<summary>Previous head was 96690a4c, new head at 1bd2f7e2. Diff between heads are:</summary>

```diff
diff --git a/src/ocean/task/IScheduler.d b/src/ocean/task/IScheduler.d
index c7d2b15e..38e1d8e4 100644
--- a/src/ocean/task/IScheduler.d
+++ b/src/ocean/task/IScheduler.d
@@ -19,7 +19,6 @@ module ocean.task.IScheduler;
 
 import core.thread;
 
-import ocean.core.Verify;
 import ocean.task.Task;
 import ocean.meta.traits.Indirections : hasIndirections;
 import ocean.meta.types.Qualifiers;
@@ -309,8 +308,7 @@ public interface IScheduler
 
 public IScheduler theScheduler ( )
 {
-    verify(_scheduler !is null,
-        "Scheduler is null, initScheduler must be called before using it");
+    assert(_scheduler !is null, "Scheduler is null, initScheduler must be called before using it");
 
     return _scheduler;
 }
diff --git a/src/ocean/task/Scheduler.d b/src/ocean/task/Scheduler.d
index c8eeaf83..0553d550 100644
--- a/src/ocean/task/Scheduler.d
+++ b/src/ocean/task/Scheduler.d
@@ -869,8 +869,7 @@ unittest
 
 public Scheduler theScheduler ( )
 {
-    verify(_scheduler !is null,
-        "Scheduler is null, initScheduler must be called before using it");
+    assert(_scheduler !is null, "Scheduler is null, initScheduler must be called before using it");
 
     return _scheduler;
 }
```
</details>

daniel-zullo commented:
Amended commit message.
